components/segmenter/src/sentence.rs

```diff
@@ -140,13 +151,70 @@ impl SentenceSegmenter {
         D: DataProvider<SentenceBreakDataV2Marker> + ?Sized,
     {
         let payload = provider.load(Default::default())?.payload;
-        Ok(Self { payload })
+        Ok(Self {
+            payload,
+            payload_locale_override: None,
+        })
+    }
+
+    icu_provider::gen_any_buffer_data_constructors!(
+        (options: SentenceBreakOptions) -> error: DataError,
+        /// Constructs a [`SentenceSegmenter`] for a given options and using compiled data.
+        ///
+        /// ✨ *Enabled with the `compiled_data` Cargo feature.*
+        ///
+        /// [📚 Help choosing a constructor](icu_provider::constructors)
+        functions: [
+            try_new_with_options,
+            try_new_with_options_with_any_provider,
+            try_new_with_options_with_buffer_provider,
+            try_new_with_options_unstable,
+            Self
+        ]
+    );
+
+    #[doc = icu_provider::gen_any_buffer_unstable_docs!(UNSTABLE, Self::try_new_with_options)]
+    pub fn try_new_with_options_unstable<D>(
+        provider: &D,
+        options: SentenceBreakOptions,
+    ) -> Result<Self, DataError>
+    where
+        D: DataProvider<SentenceBreakDataV2Marker>
+            + DataProvider<SentenceBreakDataOverrideV1Marker>
+            + ?Sized,
+    {
+        let payload = provider.load(Default::default())?.payload;
+        let payload_locale_override = if let Some(locale) = options.content_locale {
+            if locale.language == language!("el") {
```

Comment:
Nit: don't check for `el`, just try to load, and if `el` is there, it will succeed, and otherwise it will fail with `IdentifierNotFound`.